### PR TITLE
Add a couple of integration tests

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@jpkrohling the new itests are not passing yet, hence do not merge yet. If you could just have preliminary look. As promissed earlier today, I have split the the changes into two commits.

**Comment by ppalaga** [comment]
With 336cfe5 the tests should be passing. This PR is ready to be merged from my PoV.

**Comment by ppalaga** [comment]
hm... Appveyor failed. Let me have a look why...

**Comment by ppalaga** [comment]
Good that it is passing, let me make the history nicer before merging.

**Comment by ppalaga** [comment]
@jpkrohling CI passing, history prettyfied, could you please make a final review and eventually merge?

File: srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java [hunk 1]
```diff
@@ -237,6 +265,21 @@ public LocalMetadataResult find(RepositorySystemSession session, LocalMetadataRe
         return delegate.find(session, request);
     }
 
+    /**
+     * Loads the {@link Configuration} lazily. If multiple threads call this method while {@link #configurationHolder}
```
Comment (jpkrohling) [inline srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java:269]: Is this for the case the `-T` option is used, for parallel steps on the build ?

File: srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    @Rule
+    public TestRule watcher = new TestWatcher() {
+
+        @Override
+        protected void starting(Description description) {
+
+            final String rawMethodName = description.getMethodName();
+            final int bracePos = rawMethodName.indexOf('[');
+            SrcdepsLocalRepositoryManagerTest.this.currentTestName = bracePos >= 0
+                    ? rawMethodName.substring(0, bracePos) : rawMethodName;
```
Comment (jpkrohling) [inline srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java:82]: You lost me here. Care to add an example of the input and output?

File: srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                .withCliOption("-s").withCliOption(mrmSettingsXmlPath)
+        ;
+        MavenExecutionResult result = execution.execute("clean", "compile");
+        result //
+                .assertErrorFreeLog() //
+                .assertLogText(
+                        "SrcdepsLocalRepositoryManager will decorate org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManagerFactory with priority 10.0") //
+        ;
+
+        final String artifactPrefix = artifactDir + "/" + srcVersion + "/" + artifactId + "-" + srcVersion;
```
Comment (jpkrohling) [inline srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java:112]: This won't work on Windows, would it ?

File: srcdeps-maven-local-repository/src/test/projects/mvnGitBranch/pom.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.l2x6.maven.it</groupId>
+  <artifactId>it-srcdeps-branch</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  <description>A simple IT verifying the basic use case.</description>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (jpkrohling) [inline srcdeps-maven-local-repository/src/test/projects/mvnGitBranch/pom.xml:31]: Why is it here? Shouldn't it be on the parent already?

**Comment by jpkrohling** [comment]
Looks good to me, I had only a couple of minor questions, but it looks very good overall.

File: srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java (hunk 1 above)
Comment (ppalaga) [reply srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java:269]: yes, this or whatever else condition that can cause the concurrent access.

File: srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java (hunk 3 above)
Comment (ppalaga) [reply srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java:112]: AFAIK, `java.io.File` is not picky about the delimiter char: I think that both `/` and `\` work on Windows.

File: srcdeps-maven-local-repository/src/test/projects/mvnGitBranch/pom.xml (hunk 4 above)
Comment (ppalaga) [reply srcdeps-maven-local-repository/src/test/projects/mvnGitBranch/pom.xml:31]: This is a test project that does not have any parent.

File: srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java (hunk 2 above)
Comment (ppalaga) [reply srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java:82]: Added 6962104 to explain.

File: srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java (hunk 3 above)
Comment (jpkrohling) [reply srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java:112]: Oh, that's good to know!

**Comment by jpkrohling** [comment]
LGTM. Should I squash and merge, or just create a merge commit?

**Comment by ppalaga** [comment]
> LGTM. Should I squash and merge, or just create a merge commit?

I prefer a merge commit.

**Pull request merged**